src/uwtools/config.py

```diff
@@ -301,6 +301,32 @@ def update_values(self, new_dict, dict_to_update=None):
             else:
                 dict_to_update[key] = new_val
 
+    def iterate_values(self, config_dict, set_var, jinja2_var, empty_var):
+        '''
+        Given a config dictionary, recursively parse which keys are complete, which keys have 
+        unfilled jinja templates, and which keys are set to empty.  Takes in and returns a dict 
+        object and three seperate lists.
+        '''
+
+        for key, val in config_dict.items():
+            if isinstance(val, dict):
+                set_var.append(f'    {key}')
+                self.iterate_values(val, set_var, jinja2_var, empty_var)
+            elif isinstance(val, list):
+                set_var.append(f'    {key}')
+                for item in val:
+                    self.iterate_values(item, set_var, jinja2_var, empty_var)
+            elif "{{" in str(val) or "{%" in str(val):
+                jinja2_var.append(f'    {key}')
+            # elif val == "" or not val:
+            elif not val:
```

Comment:
I like the simplification here, but it will still pick up `False` and `0`. Those are legit values. Perhaps an empty string is important to distinguish. To pick up `None`, which is also important, consider using `elif val is not None`.

Reply:
I think this is what we were trying to get at when we were asking about the difference between no value provided and None/Null values

Reply:
Ahhh. Right. So I think checking explicitly for an empty string or None would be the way to go.